src/ophyd_async/panda/writers/hdf_writer.py

```diff
@@ -0,0 +1,112 @@
+import asyncio
+from typing import AsyncIterator, Dict, List, Optional
+
+from bluesky.protocols import Asset, Descriptor, Hints
+
+from ophyd_async.core import (
+    DEFAULT_TIMEOUT,
+    AsyncStatus,
+    DetectorWriter,
+    DirectoryProvider,
+    NameProvider,
+    set_and_wait_for_value,
+    wait_for_value,
+)
+
+from .panda_hdf import PandaHDF, _HDFDataset, _HDFFile
+
+
+class PandaHDFWriter(DetectorWriter):
+    def __init__(
+        self,
+        hdf: PandaHDF,
+        directory_provider: DirectoryProvider,
+        name_provider: NameProvider,
+        **scalar_datasets_paths: str,
```

Comment:
I think that we should probably check the ":CAPTURE" pv for each dataset, then only emit the `StreamResource` for the dataset if it isn't "No"

Reply:
Right, that makes sense